tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/CurriculumMembershipServiceImpl.java

```diff
@@ -57,4 +64,34 @@ public CurriculumMembershipDTO save(CurriculumMembershipDTO cmDto) {
     CurriculumMembership returnedCm = cmRepository.save(cm);
     return cmMapper.curriculumMembershipToCurriculumMembershipDto(returnedCm);
   }
+
+  /**
+   * Patch a curriculumMembership.
+   *
+   * @param cmDto the dto to patch
+   * @return the persisted object
+   */
+  @Transactional
+  public CurriculumMembershipDTO patch(CurriculumMembershipDTO cmDto)
+      throws MethodArgumentNotValidException, NoSuchMethodException {
+    log.debug("Request to patch CurriculumMembership : {}", cmDto);
+    CurriculumMembership curriculumMembershipDb = cmRepository.findById(cmDto.getId())
+        .orElseThrow(() -> new EntityNotFoundException(
+            "CurriculumMembership not found for ID: " + cmDto.getId()));
+
+    if (cmDto.getCurriculumStartDate() == null) {
+      cmDto.setCurriculumStartDate(curriculumMembershipDb.getCurriculumStartDate());
+    }
+    if (cmDto.getCurriculumEndDate() == null) {
+      cmDto.setCurriculumEndDate(curriculumMembershipDb.getCurriculumEndDate());
+    }
```

Comment:
If curriculumStartDate & curriculumEndDate are both null, actually you don't need to preceed, you can just return without any changes in the DB. Is my understanding correct?